pkg/controller/argocd/service_account_test.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 ArgoCD Operator Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package argocd
+
+import (
+	"testing"
+
+	v1 "k8s.io/api/rbac/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+func TestReconcileArgoCD_reconcileServiceAccountPermissions(t *testing.T) {
+	logf.SetLogger(logf.ZapLogger(true))
+	a := makeTestArgoCD()
+	r := makeTestReconciler(t, a)
+	_, err := r.reconcileRole("x", []v1.PolicyRule{}, a)
+	assertNoError(t, err)
+
+	assertNoError(t, r.reconcileServiceAccountPermissions("x", []v1.PolicyRule{}, a))
+}
```

Comment:
Shouldn't this make some assertions about what was reconciled?

Also, let's ensure that if it's rereconciled, it continues to work?